src/lib/util/resolveSwagger.ts

```diff
@@ -293,24 +293,26 @@ export class ResolveSwagger {
     if (!unwrappedProperty) {
       throw new Error("Null unwrapped property.")
     }
+
+    // Note: per https://editor-next.swagger.io/ tooltip when hovering over '$ref',
+    // other properties must be ignored when $ref is present:
+    //
+    //  $ref: string
+    //  Any time a subschema is expected, a schema may instead use an object containing a "$ref" property.
+    //  The value of the $ref is a URI Reference. Resolved against the current URI base,
+    //  it identifies the URI of a schema to use.
+    //  All other properties in a "$ref" object MUST be ignored.
+    //
+    parentProperty = parentProperty.$ref ? this.dereference(parentProperty.$ref) : parentProperty
+    unwrappedProperty = unwrappedProperty.$ref ? this.dereference(unwrappedProperty.$ref) : unwrappedProperty
+
     if ((!parentProperty.type || parentProperty.type === "object") && (!unwrappedProperty.type || unwrappedProperty.type === "object")) {
-      let parentPropertyToCompare = parentProperty
-      let unwrappedPropertyToCompare = unwrappedProperty
-      if (parentProperty.$ref) {
-        parentPropertyToCompare = this.dereference(parentProperty.$ref)
-      }
-      if (unwrappedProperty.$ref) {
-        unwrappedPropertyToCompare = this.dereference(unwrappedProperty.$ref)
-      }
```

Comment:
I think this is sufficient and simpler than the earlier recursive fix.  It's just moving lines 297-304 above the "if" check, and simplifying the code to update vars in-place.